File: Sample/java/src/main/java/net/nend/sample/java/banner/xmllayout/XmlSampleActivity.java
package net.nend.sample.java.banner.xmllayout;

import android.os.Bundle;

import net.nend.sample.java.R;
import net.nend.sample.java.SimpleListActivity;

import java.util.ArrayList;
import java.util.List;

public class XmlSampleActivity extends SimpleListActivity {

    private static final List<String> SAMPLE_ACTIVITIES = new ArrayList<String>() {
        {
            add(XmlLayoutActivity.class.getName());
            add(XmlResourceActivity.class.getName());
            add(XmlWithListenerActivity.class.getName());
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        instantiateMenuListFragment(R.layout.xml_sample, SAMPLE_ACTIVITIES);
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/banner/xmllayout/XmlSampleActivity.kt
package net.nend.sample.kotlin.banner.xmllayout

import android.app.Activity
import android.app.ListActivity
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.ListView
import net.nend.sample.kotlin.R

class XmlSampleActivity : ListActivity() {

    private enum class SampleType(val id: Int) {
        XML_SAMPLE(0),
        XML_RESOURCE_SAMPLE(1),
        LISTENER_SAMPLE(2);

        companion object {
            fun getType(id: Int) = values().first { it.id == id }
        }

        fun startActivity(activity: Activity) {
            when (this) {
                XML_SAMPLE -> activity.startActivity(Intent(activity,
                        XmlLayoutActivity::class.java))
                XML_RESOURCE_SAMPLE -> activity.startActivity(Intent(activity,
                        XmlResourceActivity::class.java))
                LISTENER_SAMPLE -> activity.startActivity(Intent(activity,
                        XmlWithListenerActivity::class.java))
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.xml_sample)
    }

    override fun onListItemClick(l: ListView, v: View, position: Int, id: Long) {
        super.onListItemClick(l, v, position, id)
        SampleType.getType(position).startActivity(this)
    }
}